src/xunit.performance.api/PerformanceMetric/GCAllocatedBytesForCurrentThreadEvaluator.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Parsers.MicrosoftXunitBenchmark;
+using Microsoft.Xunit.Performance.Execution;
+using Microsoft.Xunit.Performance.Sdk;
+
+namespace Microsoft.Xunit.Performance.Api
+{
+    internal class GCAllocatedBytesForCurrentThreadEvaluator : PerformanceMetricEvaluator
+    {
+        public GCAllocatedBytesForCurrentThreadEvaluator(PerformanceMetricEvaluationContext context)
+        {
+            _context = context;
+            //_context.TraceEventSource.UserData
+        }
+
+        public override void BeginIteration(TraceEvent traceEvent)
+        {
+            _bytesBeforeIteration = ((BenchmarkIterationStartArgs)traceEvent).AllocatedBytes;
+        }
+
+        public override double EndIteration(TraceEvent traceEvent)
+        {
+            var bytesAfterIteration = ((BenchmarkIterationStopArgs)traceEvent).AllocatedBytes;
```

Comment:
Rather than doing the calculation here, I think the calculation should be done before the event is written.  The written Iteration/Stop event should contain the final number so that as the harness evolves and changes are made (or the runtime stops allocating extra objects), re-processing of traces is not broken because the overhead of this feature changes.

I expect us to want to build the "re-process my trace with this new analysis" feature at some point in the future.